### ISSUE TITLE
Properties: Add syntactic support (or convention) for negation of properties

### ISSUE THREAD

**Issue by kleinj** [adl/hoaf#55]
When specifying properties, it would be convenient to be able to specify the negation of properties (e.g. not deterministic, not stutter-invariant,...).

One option would be syntactic support, e.g.,

```
properties: !stutter-invariant
```

However, this would break backward-compatibility as such an automaton would not be a valid 'HOA: v1' automaton anymore: The `!` is not in the current grammar. I don't know if that would be a big deal at the current level of tool support.

An alternative would be to consistently use a prefix such as currently used for `no-univ-branch`. There, `not-...` would probably be the most neutral from a language perspective, even though `no-...` or `non-...` might sound nicer for some of the properties.

**Comment by adl** [comment]
Assuming we use the version semantics of #56, my vote is for the `!` syntax.  I do not like the possible variations on `non-`, `not-`, `no-`, etc.  

As Jan S. suggested somewhere, we would mark `no-univ-branch` as an obsolete name for `!univ-branch`, kept to allow reading files written in v1.

**Comment by strejcek** [comment]
I've added support for negated properties to the v1.1 branch. Please check it.

**Comment by adl** [comment]
There is some discussion that started in the comments of cb2fdae about how to specify `!` that I'd like to continue here.

In the interest of clarifying negated properties, maybe we should discuss `!prop` for a few `prop` where negation is expected to be common used?  Like `!deterministic`, `!stutter-invariant`... and I was about to add `!unambiguous` when I realized this is a double negation.  Should we rename `unambiguous` as `!ambiguous`?

I think there is little point in using `!trans-acc` or `!implicit-labels` so I would not bother discussing those; yet it may makes sense to point out that `state-acc` is not the negation of `trans-acc`, and similarly that `implicit-labels` is not the negation of `explicit-labels`.

In Spot I'd also be interested in using `!inherently-weak` and `!weak`, as knowing whether an automaton is (inherently) weak or not can help you select a more efficient emptiness check, but I'm not sure this is worth discussing.

**Comment by strejcek** [comment]
I've made another pull request for this. 

I've added and explicit definition of `!deterministic`, but I don't think that an explicit note on `!stutter-invariant` is needed: the `stutter-invariant` property is already defined using a reference to some paper and the definition of `!stutter-invariant` would not say more than the negation of `stutter-invariant` holds. 

I prefer to use `unambiguous` rather than `!ambiguous` as it is the standard terminology.
